zerver/lib/message.py

```diff
@@ -1633,6 +1635,19 @@ def wildcard_mention_allowed(sender: UserProfile, stream: Stream) -> bool:
     raise AssertionError("Invalid wildcard mention policy")
 
 
+def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: List[int]) -> None:
+    user_groups = UserGroup.objects.filter(id__in=user_group_ids).select_related(
+        "can_mention_group"
+    )
+
+    for group in user_groups:
+        can_mention_group = group.can_mention_group
+        if not is_user_in_group(can_mention_group, sender, direct_member_only=False):
+            raise JsonableError(
+                _("You cannot mention user group {user_group_id}").format(user_group_id=group.id)
```

Comment:
In this case, do we want to tell them who _can_ mention them?  I guess that may be leaking information which is not-public.

Maybe `You are not allowed to` rather than `You cannot`?

Reply:
Updated the error message to use `You are not allowed to`. We are passing the `can_mention_group_id` field in the UserGroup objects sent to clients as we would require them to show the setting value in UI.

Reply:
Ah, OK.  So we _could_ tell them "You must be a member of _Cool People_ or _Awesome People_ to mention this group."

Reply:
Do we want to mention group ID or group name here? We can also change the "You cannot mention .." part to use group name instead of ID to be consistent. 

Also, do we want to mention subgroups also in "You must be a meber of ..." part in the commit message? I think it should be sufficient to just mention the supergroup and it should be undestandable that it includes subgroups as well. Also a user group can have many subgroups, so the message would become too long in some cases.

Reply:
Yeah, I think we only need to mention the supergroup - you're a member of the "Members role group" even if you're an administrator, just indirectly.

Reply:
Updated the error message to add "You must be a member of  ....." text.